src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/McCostParams.java

```diff
@@ -22,6 +26,8 @@ public class McCostParams {
   private final double[] transitReluctanceFactors;
   private final double waitReluctanceFactor;
   private final WheelchairAccessibilityRequest accessibilityRequest;
+  private final Set<FeedScopedId> unpreferredRoutes;
```

Comment:
This is ok for now, but I expect this to affect the performance. We can replace this with a BitSet when the routes and trips are indexed after the transmodel refactoring.